examples/deserialize-transaction/src/example.ts

```diff
@@ -381,7 +381,13 @@ if (identifiedInstruction === SystemInstruction.TransferSol) {
     log.info('[step 4] The first instruction calls the TransferSol instruction of the system program');
     // Narrow the type again, the instruction must have accounts to be parsed as a transfer SOL instruction
     assertIsInstructionWithAccounts(firstInstruction);
-    const parsedFirstInstruction = parseTransferSolInstruction(firstInstruction);
+
+    // TODO: This can just be `parseTransferSolInstruction(firstInstruction)` when the client is updated
+    // with the `@solana/web3.js` version that changes the instruction data type to `ReadonlyUint8Array`
+    const parsedFirstInstruction = parseTransferSolInstruction({
+        ...firstInstruction,
+        data: firstInstruction.data as unknown as Uint8Array,
+    });
```

Comment:
We'll want to create a new build of the Codama JS clients with this change to make this go away.

This was actually where I first hit this issue. I was using an API that returned instruction data as base58, and then trying to convert its output into an `IInstruction` that I could parse with Codama. Since instruction data is `Uint8Array` this required me to cast my `ReadonlyUint8Array` from encoding the API's base58 data to a `Uint8Array` like this. I figured it makes sense to just widen the type on `IInstruction` as in this PR